File: CSW/JavaScript/220909_최소원_후위표기식2.js
var fs = require('fs');
var input = fs.readFileSync(`/dev/stdin`).toString().trim().split('\n');
const N = parseInt(input[0]);
let str = input[1].trim();
const num = [];
const stack = [];


for(let i=2; i<=N+1; i++){
    num.push(parseInt(input[i]));
}

for(let i=0; i<str.length; i++){
    if(str[i]>='A'&&str[i]<='Z'){
        stack.push(num[str[i].charCodeAt()-65]);
    }else{
        let second = stack.pop();
        let first = stack.pop();
        let val = 0;
        switch(str[i]){
            case '+': 
                val = first + second; 
                stack.push(val);
                break;
            case '-': 
                val = first - second;
                stack.push(val);
                break;
            case '*': 
                val = first * second;
                stack.push(val);
                break;
            case '/': 
                val = first / second;
                stack.push(val);
                break;
        }
        
    }
}

let answer = stack.shift().toFixed(2);

console.log(answer);

File: CSW/C,C++/220909_최소원_후위표기식2.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

string str;
int N;
int num[26];
stack<double> s;

int main() {

	cin >> N >> str;

	for (int i = 0; i < N; i++) {
		cin >> num[i];
	}

	for (int i = 0; i < str.length(); i++) {
		if (str[i] >= 'A' && str[i] <= 'Z') {
			s.push(num[str[i]-'A']);
		}
		else {
			double second = s.top();
			s.pop();
			double first = s.top();
			s.pop();
			double val = 0;
			switch (str[i]) {
				case '+':
					val = first + second;
					s.push(val);
					break;
				case '-':
					val = first - second;
					s.push(val);
					break;
				case '*':
					val = first * second;
					s.push(val);
					break;
				case '/':
					val = first / second;
					s.push(val);
					break;

			}

		}
	}

	cout << fixed;
	cout.precision(2);
	cout << s.top();

	return 0;
}